dlc-manager/tests/ln_dlc_channel_execution_tests.rs

```diff
@@ -1569,34 +1569,75 @@ fn offer_sub_channel(
     }
 
     alice_node.process_events();
-    let finalize = bob_node
+    let mut finalize = bob_node
         .sub_channel_manager
         .on_sub_channel_message(&confirm, &alice_node.channel_manager.get_our_node_id())
         .unwrap()
         .unwrap();
 
-    assert_sub_channel_state!(bob_node.sub_channel_manager, channel_id, Signed);
+    assert_sub_channel_state!(bob_node.sub_channel_manager, channel_id, Finalized);
 
     bob_node.process_events();
     assert_sub_channel_state!(alice_node.sub_channel_manager, channel_id, Confirmed);
 
+    if let TestPath::Reconnect = test_path {
+        reconnect(
+            alice_node,
+            bob_node,
+            alice_descriptor.clone(),
+            bob_descriptor.clone(),
+        );
+        assert_sub_channel_state!(alice_node.sub_channel_manager, channel_id, Offered);
+        assert_sub_channel_state!(bob_node.sub_channel_manager, channel_id, Offered);
```

Comment:
👍Okay so this new branch checks that we go back to offered and proceed successfully if we were in `offerer-confirmed` and `accepter-finalized` before the reestablish.

So the code in [this thread](https://github.com/p2pderivatives/rust-dlc/pull/116/files#r1252642448) should be tested.